collector.go

```diff
@@ -70,20 +81,25 @@ type filesCollector struct {
 
 	atLeastOneCRC32Metric  bool
 	atLeastOneLineNbMetric bool
+	namespace              string
 
 	logger log.Logger
 }
 
 // Describe implements the prometheus.Collector interface.
 func (c *filesCollector) Describe(ch chan<- *prometheus.Desc) {
-	ch <- fileMatchingGlobNbDesc
-	ch <- fileSizeBytesDesc
-	ch <- fileModifTimeSecondsDesc
+	ns := c.namespace
```

Comment:
I would always take the value `file` from namespace variable as the initial value and add additional prefix if it is set.
Something like this?
```
ns := namespace
if len(c.namespace) > 0 {
  ns = namespace + "_" + ns
} 
```

Reply:
Hmm, preferring the complete namespace to be honest.